### ISSUE TITLE
SFTP user can't connect

### ISSUE THREAD

**Issue by Sebastianbuus** [Azure/azure-cli#26723]
### Describe the bug

The command to create SFTP user on storage account and regenerate password is working as intended.
The user is created and password generated in the storage account.
Password is stored for later use.

When connecting to the SFTP using a sftp client (filezilla or WinSCP) I get a connection error. "The requested container does not exist or is not accessible."

If I manually create a new sftp user on the storage account, with the same permissions and container, and then try to connect, then it works?

### Related command

``` psd1
az storage account local-user create --name $name `
  --account-name $accountName `
  --resource-group $resourceGroup `
  --home-directory $homeDirectory  `
  --has-ssh-password $hasSshPassword `
  --permission-scope permissions=$permissions service=$service resource-name=$homeDirectory

az storage account local-user regenerate-password --name $name `
  --account-name $accountName `
  --resource-group $resourceGroup
```

### Errors

Error from the SFTP client tool
"The requested container does not exist or is not accessible."

### Issue script & Debug output

DEBUG: cli.knack.cli: Command arguments: ['storage', 'account', 'local-user', 'create', '--name', '****', '--account-name', '****', '--resource-group', '****', '--home-directory', '****', '--has-ssh-password', 'True', '--has-ssh-key', 'True', '--permission-scope', 'permissions=rl', 'service=Blob', 'resource-name=****', '--debug']
DEBUG: cli.knack.cli: __init__ debug log:
Cannot enable color.
DEBUG: cli.knack.cli: Event: Cli.PreExecute []
DEBUG: cli.knack.cli: Event: CommandParser.OnGlobalArgumentsCreate [<function CLILogging.on_global_arguments at 0x0184B460>, <function OutputProducer.on_global_arguments at 0x01918BF8>, <function CLIQuery.on_global_arguments at 0x01939850>]
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreCommandTableCreate []
DEBUG: cli.azure.cli.core: Modules found from index for 'storage': ['azure.cli.command_modules.storage']
DEBUG: cli.azure.cli.core: Loading command modules:
DEBUG: cli.azure.cli.core: Name                  Load Time    Groups  Commands
DEBUG: cli.azure.cli.core: storage                   0.072        58       272
DEBUG: cli.azure.cli.core: Total (1)                 0.072        58       272
DEBUG: cli.azure.cli.core: These extensions are not installed and will be skipped: ['azext_ai_examples', 'azext_next']
DEBUG: cli.azure.cli.core: Loading extensions:
DEBUG: cli.azure.cli.core: Name                  Load Time    Groups  Commands  Directory
DEBUG: cli.azure.cli.core: Total (0)                 0.000         0         0
DEBUG: cli.azure.cli.core: Loaded 58 groups, 272 commands.
DEBUG: cli.azure.cli.core: Found a match in the command table.
DEBUG: cli.azure.cli.core: Raw command  : storage account local-user create
DEBUG: cli.azure.cli.core: Command table: storage account local-user create
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreCommandTableTruncate [<function AzCliLogging.init_command_file_logging at 0x03BFA388>]
DEBUG: cli.azure.cli.core.azlogging: metadata file logging enabled - writing logs to '****'.
INFO: az_command_data_logger: command args: storage account local-user create --name {} --account-name {} --resource-group {} --home-directory {} --has-ssh-password {} --has-ssh-key {} --permission-scope {} {} {} --debug     
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreArgumentLoad [<function register_global_subscription_argument.<locals>.add_subscription_parameter at 0x03C22460>]
DEBUG: cli.azure.cli.core.profiles._shared: Traceback (most recent call last):
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/profiles/_shared.py", line 655, in _get_attr
AttributeError: module 'azure.mgmt.storage.v2022_09_01.models' has no attribute 'ActiveDirectoryPropertiesAccountType'

DEBUG: cli.azure.cli.core.profiles._shared: Traceback (most recent call last):
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/profiles/_shared.py", line 655, in _get_attr
AttributeError: module 'azure.mgmt.storage.v2022_09_01.models' has no attribute 'ListKeyExpand'

DEBUG: cli.azure.cli.core.profiles._shared: Traceback (most recent call last):
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/profiles/_shared.py", line 655, in _get_attr
AttributeError: module 'azure.mgmt.storage.v2022_09_01.models' has no attribute 'CorsRuleAllowedMethodsItem'

DEBUG: cli.knack.cli: Event: CommandInvoker.OnPostArgumentLoad []
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPostCommandTableCreate [<function register_ids_argument.<locals>.add_ids_arguments at 0x03C38388>, <function register_cache_arguments.<locals>.add_cache_arguments at 0x03C38580>] 
DEBUG: cli.knack.cli: Event: CommandInvoker.OnCommandTableLoaded []
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreParseArgs []
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPostParseArgs [<function OutputProducer.handle_output_argument at 0x01918C40>, <function CLIQuery.handle_query_parameter at 0x01939898>, <function register_ids_argument.<locals>.parse_ids_arguments at 0x03C38538>]
DEBUG: cli.azure.cli.core.commands.client_factory: Getting management service client client_type=StorageManagementClient
DEBUG: cli.azure.cli.core.auth.persistence: build_persistence: location='****', encrypt=True
DEBUG: cli.azure.cli.core.auth.binary_cache: load: ****
DEBUG: urllib3.util.retry: Converted retries value: 1 -> Retry(total=1, connect=None, read=None, redirect=None, status=None)
DEBUG: msal.authority: openid_config = {'token_endpoint': 'https://login.microsoftonline.com/****/oauth2/v2.0/token', 'token_endpoint_auth_methods_supported': ['client_secret_post', 'private_key_jwt', 'client_secret_basic'], 'jwks_uri': 'https://login.microsoftonline.com/****/discovery/v2.0/keys', 'response_modes_supported': ['query', 'fragment', 'form_post'], 'subject_types_supported': ['pairwise'], 'id_token_signing_alg_values_supported': ['RS256'], 'response_types_supported': ['code', 'id_token', 'code id_token', 'id_token token'], 'scopes_supported': ['openid', 'profile', 'email', 'offline_access'], 'issuer': 'https://login.microsoftonline.com/****/v2.0', 'request_uri_parameter_supported': False, 'userinfo_endpoint': 'https://graph.microsoft.com/oidc/userinfo', 'authorization_endpoint': 'https://login.microsoftonline.com/****/oauth2/v2.0/authorize', 'device_authorization_endpoint': 'https://login.microsoftonline.com/****/oauth2/v2.0/devicecode', 'http_logout_supported': True, 'frontchannel_logout_supported': True, 'end_session_endpoint': 'https://login.microsoftonline.com/****/oauth2/v2.0/logout', 'claims_supported': ['sub', 'iss', 'cloud_instance_name', 'cloud_instance_host_name', 'cloud_graph_host_name', 'msgraph_host', 'aud', 'exp', 'iat', 'auth_time', 'acr', 'nonce', 'preferred_username', 'name', 'tid', 'ver', 'at_hash', 'c_hash', 'email'], 'kerberos_endpoint': 'https://login.microsoftonline.com/****/kerberos', 'tenant_region_scope': 'EU', 'cloud_instance_name': 'microsoftonline.com', 'cloud_graph_host_name': 'graph.windows.net', 'msgraph_host': 'graph.microsoft.com', 'rbac_url': 'https://pas.windows.net'}
DEBUG: msal.application: Broker enabled? False
DEBUG: cli.azure.cli.core.auth.credential_adaptor: CredentialAdaptor.get_token: scopes=('https://management.core.windows.net//.default',), kwargs={}
DEBUG: cli.azure.cli.core.auth.msal_authentication: UserCredential.get_token: scopes=('https://management.core.windows.net//.default',), claims=None, kwargs={}
DEBUG: msal.application: Found 1 RTs matching {'environment': 'login.microsoftonline.com', 'home_account_id': '15668989-6e8c-46f1-b154-7b3c88356ce8.****', 'family_id': '1'}
DEBUG: msal.telemetry: Generate or reuse correlation_id: eb4a347f-49a3-4598-92a9-c4e8867a1c5b
DEBUG: msal.application: Cache attempts an RT
DEBUG: urllib3.connectionpool: Starting new HTTPS connection (1): login.microsoftonline.com:443
DEBUG: urllib3.connectionpool: https://login.microsoftonline.com:443 "POST /****/oauth2/v2.0/token HTTP/1.1" 200 4834
DEBUG: msal.token_cache: event={
    "client_id": "04b07795-8ddb-461a-bbee-02f9e1bf7b46",
    "data": {
        "claims": "{\"access_token\": {\"xms_cc\": {\"values\": [\"CP1\"]}}}",
        "refresh_token": "********",
        "scope": [
            "https://management.core.windows.net//.default",
            "openid",
            "profile",
            "offline_access"
        ]
    },
    "environment": "login.microsoftonline.com",
    "grant_type": "refresh_token",
    "params": null,
    "response": {
        "access_token": "********",
        "client_info": "****",
        "expires_in": 4398,
        "ext_expires_in": 4398,
        "foci": "1",
        "id_token": "********",
        "scope": "https://management.core.windows.net//user_impersonation https://management.core.windows.net//.default",
        "token_type": "Bearer"
    },
    "scope": [
        "https://management.core.windows.net//user_impersonation",
        "https://management.core.windows.net//.default"
    ],
    "skip_account_creation": true,
    "token_endpoint": "https://login.microsoftonline.com/****/oauth2/v2.0/token"
}
DEBUG: cli.azure.cli.core.sdk.policies: Request URL: 'https://management.azure.com/subscriptions/****/resourceGroups/****/providers/Microsoft.Storage/storageAccounts/****/localUsers/****?api-version=2022-09-01'
DEBUG: cli.azure.cli.core.sdk.policies: Request method: 'PUT'
DEBUG: cli.azure.cli.core.sdk.policies: Request headers:
DEBUG: cli.azure.cli.core.sdk.policies:     'Content-Type': 'application/json'
DEBUG: cli.azure.cli.core.sdk.policies:     'Content-Length': '184'
DEBUG: cli.azure.cli.core.sdk.policies:     'Accept': 'application/json'
DEBUG: cli.azure.cli.core.sdk.policies:     'x-ms-client-request-id': '4e567f2a-1031-11ee-9953-3887d599c2f2'
DEBUG: cli.azure.cli.core.sdk.policies:     'CommandName': 'storage account local-user create'
DEBUG: cli.azure.cli.core.sdk.policies:     'ParameterSetName': '--name --account-name --resource-group --home-directory --has-ssh-password --has-ssh-key --permission-scope --debug'
DEBUG: cli.azure.cli.core.sdk.policies:     'User-Agent': 'AZURECLI/2.49.0 (MSI) azsdk-python-azure-mgmt-storage/21.0.0 Python/3.10.10 (Windows-10-10.0.22621-SP0)'
DEBUG: cli.azure.cli.core.sdk.policies:     'Authorization': '*****'
DEBUG: cli.azure.cli.core.sdk.policies: Request body:
DEBUG: cli.azure.cli.core.sdk.policies: {"properties": {"permissionScopes": [{"permissions": "rl", "service": "Blob", "resourceName": "****"}], "homeDirectory": "****", "hasSshKey": true, "hasSshPassword": true}} 
DEBUG: urllib3.connectionpool: Starting new HTTPS connection (1): management.azure.com:443
DEBUG: urllib3.connectionpool: https://management.azure.com:443 "PUT /subscriptions/****/resourceGroups/****/providers/Microsoft.Storage/storageAccounts/****/localUsers/****?api-version=2022-09-01 HTTP/1.1" 200 None
DEBUG: cli.azure.cli.core.sdk.policies: Response status: 200
DEBUG: cli.azure.cli.core.sdk.policies: Response headers:
DEBUG: cli.azure.cli.core.sdk.policies:     'Cache-Control': 'no-cache'
DEBUG: cli.azure.cli.core.sdk.policies:     'Pragma': 'no-cache'
DEBUG: cli.azure.cli.core.sdk.policies:     'Transfer-Encoding': 'chunked'
DEBUG: cli.azure.cli.core.sdk.policies:     'Content-Type': 'application/json'
DEBUG: cli.azure.cli.core.sdk.policies:     'Content-Encoding': 'gzip'
DEBUG: cli.azure.cli.core.sdk.policies:     'Expires': '-1'
DEBUG: cli.azure.cli.core.sdk.policies:     'Vary': 'Accept-Encoding'
DEBUG: cli.azure.cli.core.sdk.policies:     'x-ms-request-id': 'dec7fa5e-d70b-401f-90cc-c4f14609f728'
DEBUG: cli.azure.cli.core.sdk.policies:     'Strict-Transport-Security': 'max-age=31536000; includeSubDomains'
DEBUG: cli.azure.cli.core.sdk.policies:     'Server': 'Microsoft-Azure-Storage-Resource-Provider/1.0,Microsoft-HTTPAPI/2.0 Microsoft-HTTPAPI/2.0'
DEBUG: cli.azure.cli.core.sdk.policies:     'x-ms-ratelimit-remaining-subscription-writes': '1197'
DEBUG: cli.azure.cli.core.sdk.policies:     'x-ms-correlation-request-id': 'f90f5f68-0ba7-46fd-9691-07d8072b3b72'
DEBUG: cli.azure.cli.core.sdk.policies:     'x-ms-routing-request-id': 'SWEDENSOUTH:20230621T124406Z:f90f5f68-0ba7-46fd-9691-07d8072b3b72'
DEBUG: cli.azure.cli.core.sdk.policies:     'X-Content-Type-Options': 'nosniff'
DEBUG: cli.azure.cli.core.sdk.policies:     'Date': 'Wed, 21 Jun 2023 12:44:06 GMT'
DEBUG: cli.azure.cli.core.sdk.policies: Response content:
DEBUG: cli.azure.cli.core.sdk.policies: {"id":"/subscriptions/****/resourceGroups/****/providers/Microsoft.Storage/storageAccounts/****/localusers/****","name":"****","type":"Microsoft.Storage/storageAccounts/localUsers","properties":{"hasSshPassword":true,"permissionScopes":[{"permissions":"rl","service":"Blob","resourceName":"****"}],"homeDirectory":"****","sid":"S-1-2-0-2590625602-4272040639-2763996525-1021","hasSshKey":true}}
DEBUG: cli.knack.cli: Event: CommandInvoker.OnTransformResult [<function _resource_group_transform at 0x03C20BB0>, <function _x509_from_base64_to_hex_transform at 0x03C20BF8>]
DEBUG: cli.knack.cli: Event: CommandInvoker.OnFilterResult []
DEBUG: cli.knack.cli: Event: Cli.SuccessfulExecute []
DEBUG: cli.knack.cli: Event: Cli.PostExecute [<function AzCliLogging.deinit_cmd_metadata_logging at 0x03BFA4A8>]
INFO: az_command_data_logger: exit code: 0
INFO: cli.__main__: Command ran in 2.591 seconds (init: 0.551, invoke: 2.040)
INFO: telemetry.main: Begin splitting cli events and extra events, total events: 1
INFO: telemetry.client: Accumulated 0 events. Flush the clients.
INFO: telemetry.main: Finish splitting cli events and extra events, cli events: 1
INFO: telemetry.save: Save telemetry record of length 3455 in cache
INFO: telemetry.check: Returns Positive.
INFO: telemetry.main: Begin creating telemetry upload process.
INFO: telemetry.process: Creating upload process: "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\Lib\site-packages\azure\cli\telemetry\__init__.pyc C:\Users\sebuje\.azure"
INFO: telemetry.process: Return from creating process
INFO: telemetry.main: Finish creating telemetry upload process.

### Expected behavior

When connecting to the SFTP using az cli created users, I would connect in the same way, as when the SFTP user is created manually, and have no problems with the SFTP client. 

### Environment Summary

azure-cli                         2.49.0

core                              2.49.0
telemetry                          1.0.8

Extensions:
account                            0.2.5
azure-devops                      0.26.0
logic                              0.1.6
resource-graph                     2.1.0
support                            1.0.3

Dependencies:
msal                              1.20.0
azure-mgmt-resource               22.0.0

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by Sebastianbuus** [comment]
@yonzhan, any updates on this one?

**Comment by calvinhzy** [comment]
Can you run `az storage account local-user show` on both users?

**Comment by Sebastianbuus** [comment]
> @Sebastianbuus Can you run `az storage account local-user show` on both users? When you mean create another user manually, do you mean through portal?

Yes I can run `az storage account local-user show`, and `az storage account local-user list.`
All commands in az storage account local-user reference is working with out errors.

Another finding I've done is.
If I modify the user created in the script (that won't connect), and manually (through the portal) modify the Container permission;
1. Remove the container 'containername'
2. Click 'Save'
3. Add the container 'containername'
4. Add permission 'All'
5. Click 'Save'
6. Open user
7. Click 'Next'
8. Click 'Save'

Then I can connect with the same user, that didn't work earlier.

**Comment by calvinhzy** [comment]
If you see the permission given in the create request, it is only read and list, that is why I was asking if you can run the `az storage account local-user show` to see if the permission was partially missing.  
![image](https://github.com/Azure/azure-cli/assets/17182306/40f58770-5aa5-4e85-b4ec-31719c6f98bc)
![image](https://github.com/Azure/azure-cli/assets/17182306/8400e792-b7ce-4928-b913-3e4a98658330)

Can you try creating the local-user with the correct `--permission-scope permissions=rwdlc service=blob resource-name=containername`

**Comment by Sebastianbuus** [comment]
I don't know why the debug output displays that permissionscope. 'rwdlc' have been used every time.
Here is the result from `az storage account local-user show --name $name --account-name $accountName --resource-group $resourceGroup`

![image](https://github.com/Azure/azure-cli/assets/105039894/61744568-4753-49b8-a383-3eb8bbabe925)

I've tried to convert the script to Az Powershell, and I get the same issue. The scripts are running with out errors and the users and passwords are created.
When I connect with the credentials in a sFTP client it doesn't connect, unless I do the manual workaround.

Az Powershell version:

```
$permissionScope = New-AzStorageLocalUserPermissionScope -Permission rwdlc `
                                                         -Service blob `
                                                         -ResourceName $container 

Set-AzStorageLocalUser -ResourceGroupName $resourceGroup `
                       -StorageAccountName $accountName `
                       -UserName 'testuser' `
                       -PermissionScope $permissionScope `
                       -HomeDirectory $container `
                       -HasSharedKey $true -HasSshKey $true -HasSshPassword $true

New-AzStorageLocalUserSshPassword -ResourceGroupName $resourceGroup `
                                  -StorageAccountName $accountName `
                                  -UserName $name
```

**Comment by Sebastianbuus** [comment]
I've rewrote it in Azure PowerShell instead.
For some reason it is working, and Azure CLI does not

**Comment by praveenm-symplr** [comment]
We ran into the same issue today when we create the sftp user from the azure cli, not able to connect from the client and when we do manually from portal it works fine.

Double checked the permissions for the containers, password used to connect, regenerated and tried it errors out "Error: Could not connect to server". Will share the command we used in sometime and the permission scope in a while.

@yonzhan @Sebastianbuus @calvinhzy
